app/components/file_attachment_list/file_attachment_document.js

```diff
@@ -267,12 +267,18 @@ export default class FileAttachmentDocument extends PureComponent {
                                 }),
                             }]
                         );
-                        this.setStatusBarColor();
+                        this.onDonePreviewingFile();
                         RNFetchBlob.fs.unlink(path);
                     }
 
                     this.setState({downloading: false, progress: 0});
                 });
+
+                // Android does not trigger the event for DoneButtonEvent
+                // so we'll wait 4 seconds before enabling the tap for open the preview again
+                if (Platform.OS === 'android') {
+                    setTimeout(this.onDonePreviewingFile, 4000);
```

Comment:
Why do we need to do this explicitly when we are doing it irrespective of the platform above

Reply:
see description

Reply:
Of course i read the description, ticket and the previous PR before reviewing this. 
Anyway my bad. I misread the LOC in the file

This does prevent quick taps but it does not prevent it from opening again if the taps are 6sec apart or so. Android supports another listener 'RNDownloaderProgress' probably we could use this if we hit any issues with this.

I am going to approve this PR here as it is an edge case and not worth iterating.